kotlinx-coroutines-core/common/src/flow/operators/Distinct.kt

```diff
@@ -18,17 +18,36 @@ import kotlinx.coroutines.flow.unsafeFlow as flow
 @ExperimentalCoroutinesApi
 public fun <T> Flow<T>.distinctUntilChanged(): Flow<T> = distinctUntilChangedBy { it }
 
+/**
+ * Returns flow where all subsequent repetitions of the same value are filtered out, when compared
+ * with each other via the provided [areEquivalent] function.
+ */
+@FlowPreview
+public fun <T> Flow<T>.distinctUntilChanged(areEquivalent: (old: T, new: T) -> Boolean): Flow<T> =
```

Comment:
Would like to confirm this name – `distinctUntilChangedBy` feels wrong, because this overload doesn't necessarily compare "by" a single thing, and using that name also makes `distinctUntilChangedBy { it.something }` not work, because the function call is ambiguous without explicit lambda parameters. Maybe `distinctUntilChangedWith`? Or is this just fine?